lib/mongodb.js

```diff
@@ -1094,6 +1101,7 @@ MongoDB.prototype.replaceWithOptions = function(model, id, data, options, cb) {
       }
     } else {
       result = undefined;
+      idNotFoundForUpdate(id, cb);
```

Comment:
It's usually better to defer only the creation of the `Error` object to a helper method, and call the callback directly from here, to make it more clear what's happening under the hood.

Also there is no need to call `process.nextTick` at this point, because `this.execute` should have already placed our callback in one of the next ticks.

Last but not least, you must return early here, otherwise the callback is called twice (once by `idNotFoundForUpdate`, second time by `cb(err, result, cbInfo)` few lines below.

I am proposing the following:

```js
} else {
  result = undefined;
  err = errorIdNotFoundForReplace(id);
  // continue to `cb && cb(...)` below
}
```